src/hooks/user/useUserData.ts

```diff
@@ -40,19 +44,23 @@ export function useUserWithReviewsData(filters?: {
   return { loading, userData, setUserData, setUserWithReviewsFilter } as const;
 }
 
-export function useBasicUserData() {
+export function useBasicUserData(id?: number) {
   const api = useDataApi();
-
-  const loadBasicUserData = useCallback(
-    async (id: number) => {
-      return api()
+  const [loading, setLoading] = useState(true);
+  const [userData, setUserData] = useState<BasicUserDetails | null>(null);
+  useEffect(() => {
+    setLoading(true);
```

Comment:
Maybe it's better to set the loading if there is id. So just move inside the if statement because if id is null (which shouldn't happen but just for preventing infinite loading state) the loading will continue.